src/Framework/test/TargetingPackTests.cs

```diff
@@ -26,10 +27,67 @@ public TargetingPackTests(ITestOutputHelper output)
         {
             _output = output;
             _expectedRid = TestData.GetSharedFxRuntimeIdentifier();
+            _targetingPackTfm = "netcoreapp" + TestData.GetSharedFxVersion().Substring(0, 3);
             _targetingPackRoot = Path.Combine(TestData.GetTestDataValue("TargetingPackLayoutRoot"), "packs", "Microsoft.AspNetCore.App.Ref", TestData.GetTestDataValue("TargetingPackVersion"));
             _isTargetingPackBuilding = bool.Parse(TestData.GetTestDataValue("IsTargetingPackBuilding"));
         }
 
+        [Fact]
+        public void TargetingPackContainsListedAssemblies()
+        {
+            if (!_isTargetingPackBuilding)
+            {
+                return;
+            }
+
+            var actualAssemblies = Directory.GetFiles(Path.Combine(_targetingPackRoot, "ref", _targetingPackTfm), "*.dll")
+                .Select(Path.GetFileNameWithoutExtension)
+                .ToHashSet();
+            var listedTargetingPackAssemblies = TestData.ListedTargetingPackAssemblies.Keys.ToHashSet();
+
+            _output.WriteLine("==== actual assemblies ====");
+            _output.WriteLine(string.Join('\n', actualAssemblies.OrderBy(i => i)));
+            _output.WriteLine("==== expected assemblies ====");
+            _output.WriteLine(string.Join('\n', listedTargetingPackAssemblies.OrderBy(i => i)));
+
+            var missing = listedTargetingPackAssemblies.Except(actualAssemblies);
+            var unexpected = actualAssemblies.Except(listedTargetingPackAssemblies);
+
+            _output.WriteLine("==== missing assemblies from the framework ====");
+            _output.WriteLine(string.Join('\n', missing));
+            _output.WriteLine("==== unexpected assemblies in the framework ====");
+            _output.WriteLine(string.Join('\n', unexpected));
+
+            Assert.Empty(missing);
+            Assert.Empty(unexpected);
+        }
+
+        [Fact]
+        public void RefAssembliesHaveExpectedAssemblyVersions()
+        {
+            if (!_isTargetingPackBuilding)
+            {
+                return;
+            }
+
+            IEnumerable<string> dlls = Directory.GetFiles(Path.Combine(_targetingPackRoot, "ref", _targetingPackTfm), "*.dll", SearchOption.AllDirectories);
+            Assert.NotEmpty(dlls);
+
+            Assert.All(dlls, path =>
+            {
+                var fileName = Path.GetFileNameWithoutExtension(path);
+                var assemblyName = AssemblyName.GetAssemblyName(path);
+                using var fileStream = File.OpenRead(path);
+                using var peReader = new PEReader(fileStream, PEStreamOptions.Default);
+                var reader = peReader.GetMetadataReader(MetadataReaderOptions.Default);
+                var assemblyDefinition = reader.GetAssemblyDefinition();
+
+                Assert.Equal(
+                    TestData.ListedTargetingPackAssemblies[fileName],
```

Comment:
This will blow up when extra assemblies are in the targeting pack. It's sufficient that `TargetingPackContainsListedAssemblies` fails in that case. Suggest using `TryGetValue(...)` and checking its return value.